src/com/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1159,9 +1159,10 @@
     (add-certname! certname))
   (let [timestamp (to-timestamp time)
         replaced  (sql/update-values :certnames
-                                     ["name=? AND (deactivated<? OR deactivated IS NULL)" certname timestamp]
-                                     {:deactivated nil})]
-    (pos? (first replaced))))
+                                     ["name=? AND deactivated<?" certname timestamp]
```

Comment:
I think the predicate locking is actually more around the query side of the update, rather than that update itself btw, so this might be perfectly fine. Still pondering :-).